Step painter

theis999 opened the pull request:
Demo showing the effect of painting rows in the step list.

A colour picker was added to the auto put panel, mainly due to the ample amount of space there.
An event handler for updating the colour value of the colour picker is implemented.
An event handler for updating the background colour of selected rows in the step list.

To complete, the colour of rows needs to be stored in the save file.
Due note that this is built on #180

theis999 commented:
![image](https://user-images.githubusercontent.com/18309265/223416837-5e8e76ec-1f0b-43b4-b58b-1ab5f2c89997.png)

MortenTobiasNielsen commented:
I guess it cannot hurt to add this, so the user has the possibility to use it. :)

theis999 commented:
I have added **save/load** functionality

In the switch case of step parameters i also added [[[likely]]](https://en.cppreference.com/w/cpp/language/attributes/likely) which is something i stumbled upon that should optimize the compilation for those cases.

MortenTobiasNielsen commented on src/OpenTas.cpp:
```diff
@@ -163,6 +165,7 @@ Category OpenTas::extract_steps(std::ifstream& file, DialogProgressBar* dialog_p
 		step.Orientation = Capitalize(segments[5]);
 		step.Direction = Capitalize(segments[6]);
 		step.Comment = comment;
+		step.Colour = segments[10];
```
Colour won't be in the current save files, so this will crash the application.

MortenTobiasNielsen reviewed (changes requested):
You will have to handle colour in the same way as I have handled comment, so it is only extracted if it is there.

theis999 commented:
Alright, i changed some stuff and tested on file without colours, and one without colours and comments.
